src/script/miniscript.h

```diff
@@ -0,0 +1,992 @@
+                for (size_t i = diffs.size() - k; i < diffs.size(); ++i) sat_sum += diffs[i];
+                return {stat, sat_sum + dsat, dsat};
+            }
+        }
+        assert(false);
+        return {0, {}, {}};
+    }
+
+    internal::StackSize CalcStackSize() const {
+        switch (nodetype) {
+            case NodeType::PK: return {1, 1};
+            case NodeType::PK_H: return {2, 2};
+            case NodeType::OLDER: return {0, {}};
+            case NodeType::AFTER: return {0, {}};
+            case NodeType::SHA256: return {1, {}};
+            case NodeType::RIPEMD160: return {1, {}};
+            case NodeType::HASH256: return {1, {}};
+            case NodeType::HASH160: return {1, {}};
+            case NodeType::ANDOR: return {Choose(subs[0]->ss.sat + subs[1]->ss.sat, subs[0]->ss.dsat + subs[2]->ss.sat), subs[0]->ss.dsat + subs[2]->ss.dsat};
+            case NodeType::AND_V: return {subs[0]->ss.sat + subs[1]->ss.sat, {}};
+            case NodeType::AND_B: return {subs[0]->ss.sat + subs[1]->ss.sat, subs[0]->ss.dsat + subs[1]->ss.dsat};
+            case NodeType::OR_B: return {Choose(subs[0]->ss.dsat + subs[1]->ss.sat, subs[0]->ss.sat + subs[1]->ss.dsat), subs[0]->ss.dsat + subs[1]->ss.dsat};
+            case NodeType::OR_C: return {Choose(subs[0]->ss.sat, subs[0]->ss.dsat + subs[1]->ss.sat), {}};
+            case NodeType::OR_D: return {Choose(subs[0]->ss.sat, subs[0]->ss.dsat + subs[1]->ss.sat), subs[0]->ss.dsat + subs[1]->ss.dsat};
+            case NodeType::OR_I: return {Choose(subs[0]->ss.sat + 1, subs[1]->ss.sat + 1), Choose(subs[0]->ss.dsat + 1, subs[1]->ss.dsat + 1)};
+            case NodeType::THRESH_M: return {(uint32_t)keys.size() + 1, (uint32_t)keys.size() + 1};
+            case NodeType::WRAP_A: return subs[0]->ss;
+            case NodeType::WRAP_S: return subs[0]->ss;
+            case NodeType::WRAP_C: return subs[0]->ss;
+            case NodeType::WRAP_D: return {1 + subs[0]->ss.sat, 1};
+            case NodeType::WRAP_V: return {subs[0]->ss.sat, {}};
+            case NodeType::WRAP_J: return {subs[0]->ss.sat, 1};
+            case NodeType::WRAP_N: return subs[0]->ss;
+            case NodeType::JUST_1: return {0, {}};
+            case NodeType::JUST_0: return {{}, 0};
+            case NodeType::THRESH: {
+                uint32_t dsat = 0;
+                int32_t sat_sum = 0;
+                std::vector<int32_t> diffs;
+                for (const auto& sub : subs) {
+                    dsat += sub->ss.dsat.value; // The type system requires "d" for thresh, so dsat must always be valid.
+                    if (sub->ss.sat.valid) diffs.push_back((int32_t)sub->ss.sat.value - (int32_t)sub->ss.dsat.value);
+                }
+                if (diffs.size() < k) return {{}, dsat};
+                std::sort(diffs.begin(), diffs.end());
+                for (size_t i = diffs.size() - k; i < diffs.size(); ++i) sat_sum += diffs[i];
+                return {sat_sum + dsat, dsat};
```

Comment:
Looks like this is still failing on https://bitcoinbuilds.org/index.php?job=e1623137-779c-47dd-8a8a-ebaee60766ec

```
Entering test module "Bitcoin Core Test Suite"
test/miniscript_tests.cpp(282): Entering test suite "miniscript_tests"
test/miniscript_tests.cpp(284): Entering test case "fixed_tests"
test/miniscript_tests.cpp(284): Leaving test case "fixed_tests"; testing time: 35249us
test/miniscript_tests.cpp(322): Entering test case "random_tests"
script/miniscript.h:508:33: runtime error: unsigned integer overflow: 4294967295 + 11 cannot be represented in type 'unsigned int'
    #0 0x561901402481 in miniscript::Node<CPubKey>::CalcStackSize() const /home/ubuntu/src/src/./script/miniscript.h:508:33

Reply:
Fixed by switching to a different algorithm.